shell/components/toggle-button.js

```diff
@@ -57,7 +60,7 @@ class ToggleButton extends Xen.Base {
   _onToggle() {
     let {state, icons} = this._state;
     state = (state + 1) % icons.length;
-    this._setState({state});
+    //this._setState({state});
```

Comment:
Background?

Reply:
Owner of the element controls the source-of-truth for the button state (so use _props, not _state). Really should have avoided `state.state`, will add a TODO comment.